src/wp-includes/formatting.php

```diff
@@ -3161,7 +3155,42 @@ function wp_rel_callback( $matches, $rel ) {
 		}
 		$text = trim( $html );
 	}
-	return "<a $text rel=\"" . esc_attr( $rel ) . '">';
+
+	$rel_attr = $rel ? ' rel="' . esc_attr( $rel ) . '"' : '';
+
+	return "<a {$text}{$rel_attr}>";
+}
+
+/**
```

Comment:
I think that there's potential value for this in a few other places in Core. Let's remove the underscore. `wp_is_internal_url()` also feels a bit more natural to me.

With this in mind, I think it probably fits better in `functions.php` or `link-template.php`.